tools/cleanup_paths_of_violation_tables.pl

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/perl
+
+use strict;
+use warnings;
+
+use Getopt::Tabular;
+
+use NeuroDB::DBI;
+use NeuroDB::MRI;
+use NeuroDB::ExitCodes;
+
+
+
+my $profile;
+my $profile_desc = "name of config file in ../dicom-archive/.loris_mri";
+
+my @opt_table = (
+    ["-profile", "string", 1, \$profile, $profile_desc]
+);
+
+my $Help = <<HELP;
+*******************************************************************************
+Clean up paths in the violation tables
+*******************************************************************************
+
+The program replaces the invalid MincFile path present in the three violation tables
+(MRICandidateErrors, mri_violations_log and mri_protocol_violated_scans) by the
+valid path of the image present in the trashbin subdirectory of the LORIS-MRI data
+directory.
+
+Documentation: perldoc cleanup_paths_of_violation_tables.pl
+
+HELP
+my $Usage = <<USAGE;
+usage: $0 [options]
+       $0 -help to list options
+
+USAGE
+&Getopt::Tabular::SetHelp($Help, $Usage);
+&Getopt::Tabular::GetOptions(\@opt_table, \@ARGV) || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+
+
+
+##############################
+# input option error checking
+##############################
+
+if (!$ENV{LORIS_CONFIG}) {
+    print STDERR "\n\tERROR: Environment variable 'LORIS_CONFIG' not set\n\n";
+    exit $NeuroDB::ExitCodes::INVALID_ENVIRONMENT_VAR;
+}
+
+if (!defined $profile || !-e "$ENV{LORIS_CONFIG}/.loris_mri/$profile") {
+    print $Help;
+    print STDERR "$Usage\n\tERROR: You must specify a valid and existing profile.\n\n";
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+
+if ( !@Settings::db ) {
+    print STDERR "\n\tERROR: You don't have a \@db setting in the file "
+        . "$ENV{LORIS_CONFIG}/.loris_mri/$profile \n\n";
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+
+
+#########################################################
+# Establish database connection and grep config settings
+#########################################################
+
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+
+
+##################################################
+# Update the MincFile field of MRICandidateErrors
+##################################################
+
+update_MincPath_field($dbh, 'MRICandidateErrors',          'ID',    'MincFile');
+update_MincPath_field($dbh, 'mri_violations_log',          'LogID', 'MincFile');
```

Comment:
trickier for `mri_violations_log` as the file might be in the assembly directory. Need to modify the script to take that into account (joining `files` and `mri_violations_log` based on the `SeriesUID` and `EchoTime` field)

Reply:
done